jdat_notebooks/preimaging/preimaging_01_mirage.ipynb

```diff
@@ -0,0 +1,515 @@
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "path='/Users/myname/mypath'  # write here your working directory\n",
+    "\n",
+    "os.chdir(path)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "pwd"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Mirage is accompanied by a set of reference files that are used to construct the simulated data. Here we define the location of the MIRAGE data. This is the directory that contains the reference files associated with MIRAGE. \n",
+    "For users at STScI, this is the location of MIRAGE data:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "os.environ['MIRAGE_DATA'] = '/ifs/jwst/wit/mirage_data/'  # this is the STScI location of MIRAGE data"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "If the user is outside of STScI then the reference files must be downloaded using the \"downloader\" module. Please follow the instructions in https://mirage-data-simulator.readthedocs.io/en/latest/reference_files.html and create an appropriate MIRAGE_DATA location. "
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Generating input yaml files\n",
+    "\n",
+    "We begin the simulation using the programme's APT file. The xml and pointings files must be exported from APT, and are then used as input to the yaml_generator, which will generate a series of yaml input files.\n",
+    "\n",
+    "From APT we export two files: the xml and pointing files. These should be in the working directory.\n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Specify the xml and pointing files exported from APT\n",
+    "xml_file = 'NRC21_pid1069_2018_rev2.xml'\n",
+    "pointing_file = 'NRC21_pid1069_2018_rev2.pointing'\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Additional optional data to be included."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Optionally set the telescope roll angle (PAV3) for the observations\n",
+    "pav3=0.0\n",
+    "\n",
+    "# Define the output directory\n",
+    "output_dir = path"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "In this example we create NIRCam images based on a catalogue (all_filters_lmc.cat) of point sources. This catalogue contains the AB magnitude of each source in the following six filters: F070W, F150W, F200W, F277W, F356W, and F444W. "
```

Comment:
It might be worth mentioning that the dictionary of catalogs must use the APT target names as keys. Full details on yaml_generator input options are given here: https://mirage-data-simulator.readthedocs.io/en/latest/yaml_generator.html